backend/lambdas/api/repo/repo/gitlab_util/process_gitlab_utils.py

```diff
@@ -2,41 +2,37 @@
 import json
 import re
 import urllib
-from string import Template
-
 import requests
+from graphql_query import Argument, Field, Operation, Query, Variable
 
 from repo.util.aws import AWSConnect
-from repo.util.const import GITLAB_QUERY_NO_BRANCH, GITLAB_QUERY_WITH_BRANCH
 from repo.util.env import DEFAULT_ORG, REV_PROXY_DOMAIN_SUBSTRING, REV_PROXY_SECRET_HEADER
 from repo.util.utils import GetProxySecret, Logger, auth
 
 log = Logger(__name__)
 
 
-def _process_query_list(key, service_url, query_list, batch_query=True):
+def process_query_list(key, service_url, query_list, vars, batch_query=True):
     if batch_query:
-        query = "{%s}" % " ".join(query_list)
-        return _get_query_response(key, service_url, query)
+        return _get_query_response(key, service_url, query_list, vars)
     response_dict = {"data": {}}
     for query_item in query_list:
-        query = "query {}".format(query_item)
-        resp = _get_query_response(key, service_url, query)
-        if resp and "data" in resp and "project" in resp.get("data"):
-            resp_data = resp["data"]["project"]
-            repo = re.match("repo[0-9]*", query_item.strip()).group(0)
-            response_dict["data"][repo] = resp_data
+        resp = _get_query_response(key, service_url, query_item, vars)
+        if resp and "data" in resp:
+            resp_data = resp["data"]
+            repo = re.search("repo[0-9]*", query_item.strip()).group(0)
+            response_dict["data"][repo] = resp_data[repo]
```

Comment:
Can you explain this change here (in a code comment)?  I'm not familiar with GitLab's GraphQL.

Is there any chance that the `repo[0-9]*` regex might match something unexpectedly (i.e. due to using `search()` instead of `match()`)?

Reply:
I switched to search instead of match because the text is no longer at the beginning of the string. It took me awhile to figure that out.

Adding a code comment to explain better.